### ISSUE TITLE
[REVIEW]: Correlation: An Analyzing Tool for Liquids and for Amorphous Solids

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#2976]
**Submitting author:** @Isurwars (<a href="http://orcid.org/0000-0002-4359-2742">Isaías Rodríguez</a>)
**Repository:** <a href="https://github.com/Isurwars/Correlation/" target ="_blank">https://github.com/Isurwars/Correlation/</a>
**Version:** v1.0.2
**Editor:** @jgostick
**Reviewers:** @awvwgk, @CFGrote
**Archive:** <a href="https://doi.org/10.5281/zenodo.5514113" target="_blank">10.5281/zenodo.5514113</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/7d5240007263a90d59d97ab464598991/status.svg)](https://joss.theoj.org/papers/7d5240007263a90d59d97ab464598991)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/7d5240007263a90d59d97ab464598991"><img src="https://joss.theoj.org/papers/7d5240007263a90d59d97ab464598991/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/7d5240007263a90d59d97ab464598991/status.svg)](https://joss.theoj.org/papers/7d5240007263a90d59d97ab464598991)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/Isurwars/Correlation/" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@awvwgk, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @jgostick know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/Isurwars/Correlation/">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@Isurwars) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/Isurwars/Correlation/">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@Isurwars) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @awvwgk it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02976/joss.02976/10.21105.joss.02976.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02976/joss.02976/10.21105.joss.02976.pdf) :page_facing_up: :point_left:

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1103/PhysRevB.100.024422 is OK
- 10.1002/chem.200800987 is OK
- 10.3390/ma4040716 is OK
- 10.1016/j.physb.2015.07.027 is OK
- 10.1039/tf9373300027 is OK
- 10.1088/0305-4608/12/11/001 is OK
- 10.1098/rspa.1964.0147 is OK
- 10.3390/ecm-2-B001 is OK
- 10.1371/journal.pone.0147645 is OK

MISSING DOIs

- 10.1016/0025-5416(82)90073-8 may be a valid DOI for title: The structure of non-crystalline materials: liquids and amorphous solids
- 10.1063/1.2914122 may be a valid DOI for title: Models of disorder: the theoretical physics of homogeneously disordered systems

INVALID DOIs

- None
```

**Comment by Isurwars** [comment]
An initial correction to invalid arguments as well as invalid position of atoms in the input file has been submited to the testing branch (https://github.com/Isurwars/Correlation/tree/Testing). 

```
[=======================================>          ] 88 %
ERROR: The atoms:
Si_7 in position (4.07303, 4.07303, 1.35767),
Si_8 in position (4.11303, 4.07303, 1.35767),
Have a distance less than 10 pm.
```
```
[=======================================>          ] 88 %
WARNING: The atoms:
Si_7 in position (4.07303, 4.07303, 1.35767),
Si_8 in position (4.34302, 4.07303, 1.35767),
Have a distance less than the Bohr Radius.
[=======================================>          ] 98 %
Si_8 in position (4.34302, 4.07303, 1.35767),
Si_7 in position (4.07303, 4.07303, 1.35767),
Have a distance less than the Bohr Radius.
[==================================================] 100 %
Writing output files: test
```

How should i proceed?

Thanks for the feedback PhD. Sebastian Ehlert  (@awvwgk).

**Comment by jgostick** [comment]
@whedon add @CFGrote as reviewer

**Comment by whedon** [comment]
OK, @CFGrote is now a reviewer

**Comment by whedon** [comment]
:wave: @awvwgk, please update us on how your review is going.

**Comment by awvwgk** [comment]
I partly finished the review of the project in https://github.com/Isurwars/Correlation/issues/4 which lists inconsistencies in the documentation and basic blockers I found for this submission.

Also I reported a number of bugs in the command line interface (~API) of the project
- https://github.com/Isurwars/Correlation/issues/1
- https://github.com/Isurwars/Correlation/issues/2
- https://github.com/Isurwars/Correlation/issues/3

**Comment by CFGrote** [comment]
> ## Review checklist for @CFGrote
> ### Conflict of interest
> * [x]  I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.
> 
> ### Code of Conduct
> * [x]  I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).
> 
> ### General checks
> * [ ]  **Repository:** Is the source code for this software available at the [repository url](https://github.com/Isurwars/Correlation/)?
> * [ ]  **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
> * [ ]  **Contribution and authorship:** Has the submitting author (@Isurwars) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
> * [ ]  **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)
> 
> ### Functionality
> * [ ]  **Installation:** Does installation proceed as outlined in the documentation?
> * [ ]  **Functionality:** Have the functional claims of the software been confirmed?
> * [ ]  **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)
> 
> ### Documentation
> * [ ]  **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
> * [ ]  **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
> * [ ]  **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
> * [ ]  **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
> * [ ]  **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
> * [ ]  **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support
> 
> ### Software paper
> * [ ]  **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
> * [ ]  **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
> * [ ]  **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
> * [ ]  **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
> * [ ]  **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax](https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by CFGrote** [comment]
@whedon help

**Comment by whedon** [comment]
I'm sorry human, I don't understand that. You can see what commands I support by typing:

```
@whedon commands
```

**Comment by CFGrote** [comment]
@whedon commands

**Comment by whedon** [comment]
Here are some things you can ask me to do:

```
# List Whedon's capabilities
@whedon commands

# List of editor GitHub usernames
@whedon list editors

# List of reviewers together with programming language preferences and domain expertise
@whedon list reviewers

EDITORIAL TASKS

# Compile the paper
@whedon generate pdf

# Compile the paper from alternative branch
@whedon generate pdf from branch custom-branch-name

# Ask Whedon to check the references for missing DOIs
@whedon check references

# Ask Whedon to check repository statistics for the submitted software
@whedon check repository

```

**Comment by CFGrote** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02976/joss.02976/10.21105.joss.02976.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02976/joss.02976/10.21105.joss.02976.pdf) :page_facing_up: :point_left:

**Comment by CFGrote** [comment]
Review comments are in https://github.com/Isurwars/Correlation/issues/7.

**Comment by jgostick** [comment]
@awvwgk, how is your review coming along?  I see you checked a few boxes.  Are you stuck?  Do the authors needs to update their package somehow?

**Comment by awvwgk** [comment]
I finished my review of the software paper in https://github.com/Isurwars/Correlation/issues/8, the issues regarding the software / documentation / testing are mentioned here https://github.com/openjournals/joss-reviews/issues/2976#issuecomment-772875660.

**Comment by jgostick** [comment]
@awvwgk I'm sorry!  I missed your comment above in the background noise.  Thanks for your detailed efforts.

**Comment by jgostick** [comment]
@Isurwars You've been quite silent on this thread.  Are you busy addressing the reviewers comments?

**Comment by jgostick** [comment]
@Isurwars are you there?

**Comment by jgostick** [comment]
I have reached out the authors via email addresses and other means, but have not been able to track them down.  I'm not sure how long a submission can sit waiting before we give up?

**Comment by danielskatz** [comment]
@jgostick and @lsuwars - let's wait another 3 weeks, and if there is no contact, we can reject this, given where it is in the process, though it could be resubmitted.  I'll mark it as paused for now.

**Comment by jgostick** [comment]
Soooo, someone from this group just reached out to me via email.  I have sent them to this link and told them to address the reviewer comments....some of which were actually posted to their repo directly instead of using this forum/checklist.

**Comment by danielskatz** [comment]
Ok - I hope we will see some activity soon, or we will go ahead and reject this, but it sounds like maybe that won't be necessary.

**Comment by Isurwars** [comment]
Sorry for the delay in the answer.

Installation
•	There is no configure step involved which means if I had non-standard settings on my computer, make might fail to locate the compiler and/or libraries. Also, installing into a target directory is not supported (no install target).

•	We make a configure script with the help of GNU autotools. The installation process s followed now by the standard ./configure, make, make install. The old Makefile is included in case of incompatibility with the new GNU autotools option.

•	You could explain why the user should run make clean after building.

•	The command make clean deletes object files created during the compilation process, and is not required to be run after compilation, it is used to keep a clean folder, and it is only recommended to clean the mess for GitHub upload. This info has been added to the README.md file.

•	The binary produced by the make all step has the windows-typical '.exe' extension. This is a bit unusual for linux/mac.

•	You are right. The ‘.app’ extension should be used in MacOS; as far as we know there is no standard extension for executables in linux, so no extension should be used. The new Makefile automatically adds the system extension to the executable. 
Functionality
•	make tests produces no errors, however it remains unclear if that just means the program ends or if it actually passed unittest and/or regression tests. Probably related to #5.

•	This has been fixed in the testing branch. Currently the testing branch displays a “successfully calculated.” message at the end, as well as warnings and errors.

Running the individual test with ./bin/correlation.exe test/test_1/aPd.cell
returns "File not found". aPd.cell is in test_3, test run is ok.
•	Tests were rearranged to coincide with the paper numerals.

•	Plotting the resulting PDF from test_3 gives the following figure: 
 .
You might consider putting this or a similar graph into the README and instruct the user to try and reproduce this result as a confirmation the code works as it should.
•	Thanks for the recommendation, we added images for the test and csv files.
•	Performance: You claim that your code can handle up to 25000 atoms. I could not find a demonstration of such a system. Is it in test_6? If so, you should clearly mention it.

•	This test was removed from the Release Candidate 1 (RC1) test samples because it required more than 15 minutes and more than 8 GB in RAM, after the optimization introduced in RC2, this test was reintroduced as test_7.
Documentation
•	I find the output not properly documented (neither in the paper nor in the README): Describe the content of each file, describe what each column contains and specify the units. At a minimum, this information should go into the header of each .csv file. As a perspective, I recommend storing the results in a self-documenting format, e.g. hdf5.

•	I agree. We have added documentation in the README.md file. The support for hdf5 file as well as for other formats should be added in a future release.

•	You should document the command line options in the README.

•	I agree. See above.

•	No automated tests, see comment above. Adding tests that compare the code's output to reference data will make the validity of your code more accessible.

•	A complete reworking of the Makefile and of the installation process was made in RC2 and RC3. The documentation of the new automated processes was included in RC4.

•	No community guidelines (probably related to #6)

•	Paper
•	Summary: 
o	You state that the code is ready for mac while the README says mac not supported due to compiler issues.

o	Corrected in RC4.

o	A non-expert reader will be overwhelmed by the technical and field specific jargon. I advise to write something like: "Our code calculates various correlation functions from atomic positions".

o	We included your phrase in the abstract. I hope that is enough.

o	Your claimed limit of 25000 atoms is not a lot (e.g. in biomolecules). What causes this limit? Does it depend on hardware/software on the user's system?

o	This limit was caused by the algorithm used to calculate the correlations in the system. In RC1 we switched from an all vs all paradigm to a near-neighbors paradigm, effectively reducing the computing time by an order of magnitude and simultaneously the RAM requirements were reduced by two orders of magnitude. 
This change has improved the code much, so now samples of 100,000+ atoms can be handled even in a laptop with 16 GB of RAM in less than 10 minutes instead of requiring 64+ GB of RAM and a Workstation.
•	
•	Statement of need: 
o	name and cite other packages (earlier, in statement of need). Please check the exhaustive list of software in the Computational Physics Communications archive http://cpc.cs.qub.ac.uk/, the IUCr's software list https://www.iucr.org/resources/other-directories/software for similar software. What is particular about your software that makes it stand out from other packages?clearly name target audience


•	Benchmarks, validation: 
o	define test_x as being included in the software package. a table with
references would be helpful.

o	I agree; the table is included in the README.md file.

o	Is test_6 really a challenge? it completes on my laptop in a few
seconds. What are the limiting factors? memory/cpu? What exactly do you mean
by "... benchmark memory and CPU performance ...". The sentence on line 91-93 mentions test_6 and test_4, it is not clear which one you refer to here.

o	Test 7 is now included; it should require about 10 minutes and more than 4 GB of RAM to compute with the new algorithm in a regular laptop.

•	Conclusion: 
o	You mention HPC here but HTC (high throughput computing) earlier. What exactly do you mean here? Does your code support parallelization using e.g. MPI or similar frameworks? Suppose I want to run a large system that does not fit into memory, can your code deal with it? How do you address parallel IO?

o	HPC vs HTC corrected. Currently the code does not support MPI parallelization; this will be a top priority in the future. My code cannot deal with a system that does not fit in to memory. My code is designed to work with several structures in parallel, as required by Machine Learning (ML) algorithms and because of the modular output the code can easily be integrated in an ML workflow.

o	How will it be extended, which features are missing?

o	The support for different kinds of output files like hdf5 standard is one of the top priorities in the near future.
o	Also, the inclusion of other correlation functions like Velocity Correlation Functions to further improve the analysis of liquids and phase transitions is the other top priority. 
o	Finally, the parallelization of the main loop could further improve the code by switching to a “divide-and-conquer” paradigm.

**Comment by jgostick** [comment]
@lsurwars, you have done an admirable job in responding to the reviewer comments.  I will ping them via email to have them revisit their check lists, in case they're not being notified by github.

**Comment by awvwgk** [comment]
@Isurwars thanks for addressing our comments. Personally, I would prefer to discuss those points in the issue tracker of your projects as it allows to more clearly associate changes or patches with an already raised topic, not only for us, but also for your users, but we can also have this discussion here.

**Comment by CFGrote** [comment]
Hi there, just coming back from vacation, I'll do another review as soon as I digged through the pileup on my desk ...

**Comment by CFGrote** [comment]
@Isurwars, I can't find some of the changes you list in your review response (such as the configure script) in the main branch. Shall I wait until you pushed everything to main?

**Comment by CFGrote** [comment]
@whedon generate pdf from branch Testing

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch Testing. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02976/joss.02976/10.21105.joss.02976.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02976/joss.02976/10.21105.joss.02976.pdf) :page_facing_up: :point_left:

**Comment by CFGrote** [comment]
My 2nd round of review comments are here: https://github.com/Isurwars/Correlation/issues/7#issuecomment-884233704 with links to issues for each comment.

**Comment by awvwgk** [comment]
I updated the issues at the repo and closed everything that is already addressed. Some issues are still unaddressed or only partially resolved.

**Comment by jgostick** [comment]
Hi @lsurwars, @CFGrote only has 2 'unchecked boxes' in his list above, and they both seem addressable.  Can you finish these off?
- [ ] State of the field: Do the authors describe how this software compares to other commonly-used packages?
- [ ] References: Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper citation syntax?

**Comment by jgostick** [comment]
Also, @Isurwars,  there are a few additional points that @CFGrote has posted on your repo.  Once these are addressed, we'll be able to move forward.  Almost there!

**Comment by jgostick** [comment]
@awvwgk, is your checklist at the top of this thread up-to-date?  Do they outline your points about which things are unaddressed?

**Comment by awvwgk** [comment]
Looks good to me now. I have only minor thing left unaddressed, but those don't hinder moving the paper forward (still would be nice if they get addressed). The only outstanding issue in my opinion are the missing contributing guidelines for the project (see https://github.com/Isurwars/Correlation/issues/6).

**Comment by jgostick** [comment]
Hi @Isurwars, have you addressed the remaining 2 comments by @CFGrote?  

- State of the field: Do the authors describe how this software compares to other commonly-used packages?
- References: Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper citation syntax?

**Comment by jgostick** [comment]
Hi @Isurwars, @awvwgk has requested that you address one last item:

- [ ] Community guidelines: Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

Can you follow through with this then let them and I know, so we can complete the review checklist and move on to the next stage?

**Comment by awvwgk** [comment]
> [ ] Community guidelines: Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

Has been addressed now with https://github.com/Isurwars/Correlation/commit/7da2042083119b88d5f2c6c7d6ec21b70fcca241

**Comment by jgostick** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02976/joss.02976/10.21105.joss.02976.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02976/joss.02976/10.21105.joss.02976.pdf) :page_facing_up: :point_left:

**Comment by jgostick** [comment]
wow, this one took a while, but it seems that it's ready to be accepted.  Both reviewers are fully satisfied.  (Note that @CFGrote and the author did most of their exchanges in the issues tracker of the official package).  I will begin moving this through the acceptance stages.  @Isurwars, you'll have a bit more to do, such as create a doi at zenodo.  I'll make a checklist below for you. Congratulations.

**Comment by jgostick** [comment]
- [ ] Double check authors and affiliations (including ORCIDs)
- [ ] Make a release of the software with the latest changes from the review and post the version number here. This is the version that will be used in the JOSS paper.
- [ ] Archive the release on Zenodo/figshare/etc and post the DOI here.
- [ ] Make sure that the title and author list (including ORCIDs) in the archive match those in the JOSS paper.

**Comment by Isurwars** [comment]
>     * [x]  Double check authors and affiliations (including ORCIDs)
The affiliations have been updated, and most of the ORCIDs have been added,  Professor Alexander Valladares does not have an ORCID.
 
>     * [x]  Make a release of the software with the latest changes from the review and post the version number here. This is the version that will be used in the JOSS paper.

The version used for the paper is v1.0.0, https://github.com/Isurwars/Correlation/releases/tag/V1.0.0

 
>     * [x]  Archive the release on Zenodo/figshare/etc and post the DOI here.  Following [this guide](https://guides.github.com/activities/citable-code/) if needed.

The version has been archived in zenodo: https://zenodo.org/record/5499455
With DOI:  10.5281/zenodo.5499455

>     * [x]  Make sure that the title and author list (including ORCIDs) in the archive match those in the JOSS paper.

Author's affiliations and ORCIDs have been updated in the zenodo release.

**Comment by jgostick** [comment]
@whedon set V1.0.0 as version

**Comment by whedon** [comment]
OK. V1.0.0 is the version.

**Comment by jgostick** [comment]
@whedon set 10.5281/zenodo.5499455 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.5499455" target="_blank">10.5281/zenodo.5499455</a> is the archive.

**Comment by jgostick** [comment]
@whedon accept

**Comment by whedon** [comment]
To recommend a paper to be accepted use `@whedon recommend-accept`

**Comment by jgostick** [comment]
@whedon recommend-accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1103/PhysRevB.100.024422 is OK
- 10.1002/chem.200800987 is OK
- 10.3390/ma4040716 is OK
- 10.1016/j.physb.2015.07.027 is OK
- 10.1039/tf9373300027 is OK
- 10.1088/0305-4608/12/11/001 is OK
- 10.1098/rspa.1964.0147 is OK
- 10.3390/ecm-2-B001 is OK
- 10.1371/journal.pone.0147645 is OK
- 10.1021/ct400341p is OK

MISSING DOIs

- 10.1016/0025-5416(82)90073-8 may be a valid DOI for title: The structure of non-crystalline materials: liquids and amorphous solids
- 10.1063/1.2914122 may be a valid DOI for title: Models of disorder: the theoretical physics of homogeneously disordered systems

INVALID DOIs

- https://doi.org/10.1016/j.cpc.2021.108033 is INVALID because of 'https://doi.org/' prefix
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2584

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2584, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by jgostick** [comment]
@Isurwars, can you fix the DOIs in your references as noted by whedon above?

**Comment by Isurwars** [comment]
I have fixed the DOIs and created a new release v1.0.1:

https://github.com/Isurwars/Correlation/releases/tag/v1.0.1

I also create a corresponding zenodo release:

https://zenodo.org/record/5501541

**Comment by jgostick** [comment]
@whedon set V1.0.1 as version

**Comment by whedon** [comment]
OK. V1.0.1 is the version.

**Comment by jgostick** [comment]
@whedon set 10.5281/zenodo.5501541 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.5501541" target="_blank">10.5281/zenodo.5501541</a> is the archive.

**Comment by jgostick** [comment]
@whedon recommend-accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1103/PhysRevB.100.024422 is OK
- 10.1002/chem.200800987 is OK
- 10.3390/ma4040716 is OK
- 10.1016/j.physb.2015.07.027 is OK
- 10.1039/tf9373300027 is OK
- 10.1088/0305-4608/12/11/001 is OK
- 10.1098/rspa.1964.0147 is OK
- 10.3390/ecm-2-B001 is OK
- 10.1371/journal.pone.0147645 is OK
- 10.1016/0025-5416(82)90073-8  is OK
- 10.1063/1.2914122 is OK
- 10.1021/ct400341p is OK
- 10.1016/j.cpc.2021.108033 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
PDF failed to compile for issue #2976 with the following error: 
```
 /app/vendor/bundle/ruby/2.6.0/gems/rest-client-2.1.0/lib/restclient/request.rb:749:in `rescue in transmit': Timed out connecting to server (RestClient::Exceptions::OpenTimeout)
	from /app/vendor/bundle/ruby/2.6.0/gems/rest-client-2.1.0/lib/restclient/request.rb:651:in `transmit'
	from /app/vendor/bundle/ruby/2.6.0/gems/rest-client-2.1.0/lib/restclient/request.rb:163:in `execute'
	from /app/vendor/bundle/ruby/2.6.0/gems/rest-client-2.1.0/lib/restclient/request.rb:63:in `execute'
	from /app/vendor/bundle/ruby/2.6.0/gems/rest-client-2.1.0/lib/restclient.rb:66:in `get'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-c5c16aedb3d6/lib/whedon/compilers.rb:119:in `pdf_from_markdown'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-c5c16aedb3d6/lib/whedon/compilers.rb:13:in `generate_pdf'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-c5c16aedb3d6/lib/whedon/processor.rb:99:in `compile'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-c5c16aedb3d6/bin/whedon:88:in `compile'
	from /app/vendor/bundle/ruby/2.6.0/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
	from /app/vendor/bundle/ruby/2.6.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
	from /app/vendor/bundle/ruby/2.6.0/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
	from /app/vendor/bundle/ruby/2.6.0/gems/thor-0.20.3/lib/thor/base.rb:466:in `start'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-c5c16aedb3d6/bin/whedon:131:in `<top (required)>'
	from /app/vendor/bundle/ruby/2.6.0/bin/whedon:23:in `load'
	from /app/vendor/bundle/ruby/2.6.0/bin/whedon:23:in `<main>'
/app/vendor/ruby-2.6.6/lib/ruby/2.6.0/net/http.rb:947:in `initialize': execution expired (Net::OpenTimeout)
	from /app/vendor/ruby-2.6.6/lib/ruby/2.6.0/net/http.rb:947:in `open'
	from /app/vendor/ruby-2.6.6/lib/ruby/2.6.0/net/http.rb:947:in `block in connect'
	from /app/vendor/ruby-2.6.6/lib/ruby/2.6.0/timeout.rb:103:in `timeout'
	from /app/vendor/ruby-2.6.6/lib/ruby/2.6.0/net/http.rb:945:in `connect'
	from /app/vendor/ruby-2.6.6/lib/ruby/2.6.0/net/http.rb:930:in `do_start'
	from /app/vendor/ruby-2.6.6/lib/ruby/2.6.0/net/http.rb:919:in `start'
	from /app/vendor/bundle/ruby/2.6.0/gems/rest-client-2.1.0/lib/restclient/request.rb:727:in `transmit'
	from /app/vendor/bundle/ruby/2.6.0/gems/rest-client-2.1.0/lib/restclient/request.rb:163:in `execute'
	from /app/vendor/bundle/ruby/2.6.0/gems/rest-client-2.1.0/lib/restclient/request.rb:63:in `execute'
	from /app/vendor/bundle/ruby/2.6.0/gems/rest-client-2.1.0/lib/restclient.rb:66:in `get'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-c5c16aedb3d6/lib/whedon/compilers.rb:119:in `pdf_from_markdown'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-c5c16aedb3d6/lib/whedon/compilers.rb:13:in `generate_pdf'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-c5c16aedb3d6/lib/whedon/processor.rb:99:in `compile'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-c5c16aedb3d6/bin/whedon:88:in `compile'
	from /app/vendor/bundle/ruby/2.6.0/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
	from /app/vendor/bundle/ruby/2.6.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
	from /app/vendor/bundle/ruby/2.6.0/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
	from /app/vendor/bundle/ruby/2.6.0/gems/thor-0.20.3/lib/thor/base.rb:466:in `start'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-c5c16aedb3d6/bin/whedon:131:in `<top (required)>'
	from /app/vendor/bundle/ruby/2.6.0/bin/whedon:23:in `load'
	from /app/vendor/bundle/ruby/2.6.0/bin/whedon:23:in `<main>'

```

**Comment by Isurwars** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02976/joss.02976/10.21105.joss.02976.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02976/joss.02976/10.21105.joss.02976.pdf) :page_facing_up: :point_left:

**Comment by Kevin-Mattheus-Moerman** [comment]
@Isurwars 

To process this work for acceptance there are some minor last steps to work on: 
- [ ] Please edit (you have to do this manually) the Zenodo archive DOI such that the title matches that of the paper: `Correlation: An Analysis Tool for Liquids and for Amorphous Solids`

I have read the paper and have some minor comments below: 
- [ ] Check additional space in `(rdfpy, RadialDistributionFunction )` which should be `(rdfpy, RadialDistributionFunction)`
- [ ] Just a suggestion, not required: The mathematical notation for the sine function in equation 1 seems italic. Perhaps use `\sin` in Latex so it is rendered in non-italic like a function.  Again this is a rather minor suggestion. 
- [ ] You may choose to use bold writing for software names e.g. **Correlation**, however it is more common in JOSS if they are written in code script e.g. `Correlation`. This is just a suggestion, not required.

**Comment by Isurwars** [comment]
* [x]  Please edit (you have to do this manually) the Zenodo archive DOI such that the title matches that of the paper: `Correlation: An Analysis Tool for Liquids and for Amorphous Solids`

Done.


* [x]  Check additional space in `(rdfpy, RadialDistributionFunction )` which should be `(rdfpy, RadialDistributionFunction)`

Removed space in the paper.md
 
* [x]  Just a suggestion, not required: The mathematical notation for the sine function in equation 1 seems italic. Perhaps use `\sin` in Latex so it is rendered in non-italic like a function.  Again this is a rather minor suggestion.

Changed from:
```
\begin{equation}\label{eq:PDF}
g(r) - 1= \frac{1}{2 \pi^2 \rho*0 r}\int*{0}^{\infty} Q[S(Q)-1]sin(Qr)dQ,
\end{equation}
```
To:

```
\begin{equation}\label{eq:PDF}
g(r) - 1= \frac{1}{2 \pi^2 \rho*0 r}\int*{0}^{\infty} Q[S(Q)-1]\sin(Qr)dQ,
\end{equation}
```

* [x]  You may choose to use bold writing for software names e.g. **Correlation**, however it is more common in JOSS if they are written in code script e.g. `Correlation`. This is just a suggestion, not required.


The corrections were added in v1.0.2: https://github.com/Isurwars/Correlation/releases/tag/v1.0.2

And the new zenodo archive: https://zenodo.org/record/5514113

**Comment by arfon** [comment]
@whedon set 10.5281/zenodo.5514113 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.5514113" target="_blank">10.5281/zenodo.5514113</a> is the archive.

**Comment by arfon** [comment]
@whedon set v1.0.2 as version

**Comment by whedon** [comment]
OK. v1.0.2 is the version.

**Comment by arfon** [comment]
@whedon recommend-accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1103/PhysRevB.100.024422 is OK
- 10.1002/chem.200800987 is OK
- 10.3390/ma4040716 is OK
- 10.1016/j.physb.2015.07.027 is OK
- 10.1039/tf9373300027 is OK
- 10.1088/0305-4608/12/11/001 is OK
- 10.1098/rspa.1964.0147 is OK
- 10.3390/ecm-2-B001 is OK
- 10.1371/journal.pone.0147645 is OK
- 10.1016/0025-5416(82)90073-8 is OK
- 10.1063/1.2914122 is OK
- 10.1021/ct400341p is OK
- 10.1016/j.cpc.2021.108033 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2591

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2591, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by arfon** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1439599053482991620) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/2592
1. Wait a couple of minutes, then verify that the paper DOI resolves [https://doi.org/10.21105/joss.02976](https://doi.org/10.21105/joss.02976)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by arfon** [comment]
@awvwgk, @CFGrote – many thanks for your reviews here and to @jgostick  for editing this submission! JOSS relies upon the volunteer effort of people like you and we simply wouldn't be able to do this without you ✨ 

@Isurwars – your paper is now accepted and published in JOSS :zap::rocket::boom:

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.02976/status.svg)](https://doi.org/10.21105/joss.02976)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.02976">
  <img src="https://joss.theoj.org/papers/10.21105/joss.02976/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.02976/status.svg
   :target: https://doi.org/10.21105/joss.02976
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.02976/status.svg)](https://doi.org/10.21105/joss.02976)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss
